src/api/app/controllers/webui/request_controller.rb

```diff
@@ -515,53 +540,39 @@ def staging_status(request, target_project)
+  end
+
+  def header_data
+    target_project = Project.find_by_name(@bs_request.target_project_name)
+    @staging_status = staging_status(@bs_request, target_project) if Staging::Workflow.find_by(project: target_project)
 
     @diff_limit = params[:full_diff] ? 0 : nil
     @diff_to_superseded_id = params[:diff_to_superseded]
-
-    # Handling request actions
     @action = @bs_request.webui_actions(filelimit: @diff_limit, tarlimit: @diff_limit, diff_to_superseded: @diff_to_superseded,
                                         diffs: true, action_id: @action_id.to_i, cacheonly: 1).first
-    active_action_index = @supported_actions.index(@active_action)
-    if active_action_index
-      @prev_action = @supported_actions[active_action_index - 1] unless active_action_index.zero?
-      @next_action = @supported_actions[active_action_index + 1] if active_action_index + 1 < @supported_actions.length
-    end
-
-    target_project = Project.find_by_name(@bs_request.target_project_name)
-    @request_reviews = @bs_request.reviews.for_non_staging_projects(target_project)
-    @staging_status = staging_status(@bs_request, target_project) if Staging::Workflow.find_by(project: target_project)
-    @refresh = @action[:diff_not_cached]
 
-    # Collecting all issues in a hash. Each key is the issue name and the value is a hash containing all the issue details.
-    @issues = @action.fetch(:sourcediff, []).reduce({}) { |accumulator, sourcediff| accumulator.merge(sourcediff.fetch('issues', {})) }
+    active_action_index = @supported_actions.index(@active_action)
 
-    # retrieve a list of all package maintainers that are assigned to at least one target package
-    @package_maintainers = target_package_maintainers
+    return unless active_action_index
 
-    # retrieve a list of all project maintainers
-    @project_maintainers = target_project&.maintainers || []
+    @prev_action = @supported_actions[active_action_index - 1] unless active_action_index.zero?
+    @next_action = @supported_actions[active_action_index + 1] if active_action_index + 1 < @supported_actions.length
+  end
 
-    # search for a project, where the user is not a package maintainer but a project maintainer and show
-    # a hint if that package has some package maintainers (issue#1970)
-    @show_project_maintainer_hint = !@package_maintainers.empty? && @package_maintainers.exclude?(User.session) && any_project_maintained_by_current_user?
+  def tabs_data
+    # Collecting all issues in a hash. Each key is the issue name and the value is a hash containing all the issue details.
 
-    # Handling build results
-    @staging_project = @bs_request.staging_project.name unless @bs_request.staging_project_id.nil?
+    @issues = @action.fetch(:sourcediff, []).reduce({}) { |accumulator, sourcediff| accumulator.merge(sourcediff.fetch('issues', {})) }
     @actions_for_diff = [:submit, :delete, :maintenance_incident, :maintenance_release]
```

Comment:
@ncounter initializing this variable `@actions_for_diff` in `tabs_data` , and then using it in some of the other views/controller methods is IMO very implicit. Like I get the impression its just needed for tabs, but its not. Maybe we find a way to improve it

Reply:
Well, actually `@actions_for_diff` is used by other views but only to pass it to the `request_tabs` partial, therefore I defined a method where to instantiate variables required by that partial view. It is more odd to me (and I am with you on this) about the `@issues` variable which is initialized in `tabs_data` but it also used in the `mentioned_issues` not only for tabs but also for the content. In this case I agree, it is a bit implicit. But I'd also say it is a common pattern with instance variables (think about `@bs_request`, `@action`, etc.).

Btw, overall I agree and I am open to adapt it.
Do you think something like a _singleton pattern_ would work better? Something like:
```ruby
def get_issues
  return @issues if defined?(@issues)

  @issues = @action.fetch(:sourcediff, []).reduce({}) { |accumulator, sourcediff| accumulator.merge(sourcediff.fetch('issues', {})) }
end
```
and then use the method wherever the instance variable is meant to be fetched.

Reply:
What about `[:submit, :delete, :maintenance_incident, :maintenance_release]` is variable? :)